pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -4195,34 +4195,40 @@ pj_status_t pjsua_acc_update_contact_on_ip_change(pjsua_acc *acc)
     if ((status != PJ_SUCCESS)
         && (acc->ip_change_op == PJSUA_IP_CHANGE_OP_ACC_UPDATE_CONTACT))
     {
-	if (status == PJSIP_EBUSY) {
-	    /* Retry registration by cancelling the existing one. */
-	    if (acc->regc) {
-		destroy_regc(acc, PJ_TRUE);
-		update_keep_alive(acc, PJ_FALSE, NULL);
-
-		status = pjsua_acc_set_registration(acc->index, PJ_TRUE);
-		if (status == PJ_SUCCESS) {
-		    return status;
-		}
-	    }
-	}
-
-	if (pjsua_var.ua_cfg.cb.on_ip_change_progress) {
-	    /* If update contact fails, notification might already been
-	     * triggered from registration callback.
-	     */
-	    pjsua_ip_change_op_info info;
-
-	    pj_bzero(&info, sizeof(info));
-	    info.acc_update_contact.acc_id = acc->index;
-	    info.acc_update_contact.is_register = !need_unreg;
-
-	    pjsua_var.ua_cfg.cb.on_ip_change_progress(acc->ip_change_op,
-						      status,
-						      &info);
-	}
-	pjsua_acc_end_ip_change(acc);
+        if (status == PJSIP_EBUSY) {
+            /* Retry the (un)registration. */
+            if (acc->regc) {
+                status = PJ_SUCCESS;
+                destroy_regc(acc, PJ_TRUE);
+                update_keep_alive(acc, PJ_FALSE, NULL);
+
+                if (need_unreg)
+                    status = pjsua_regc_init(acc->index);
```

Comment:
If the regc is destroyed in `destroy_regc()` (where the contact.slen seems to be reset), will the retry try to unregister the new contact instead of the old contact?

Reply:
handle this in the last commit